[18.0][MIG] delivery_carrier_label_batch: Migration to 18.0

yankinmax opened the pull request:
Depends on:
- [ ] https://github.com/OCA/delivery-carrier/pull/900
- [ ] https://github.com/OCA/delivery-carrier/pull/974
- [ ] https://github.com/OCA/stock-logistics-workflow/pull/1855

yvaucher committed:
[9.0] port module delivery_carrier_label_dispatch

- change name to batch to match with change of `picking_dispatch` name to
`stock_batch_picking`
- mode description in README.rst
- move files in models and views directories
- change headers to small license header
- use api 8.0
- fix and improve tests

yvaucher committed:
[9.0] port module delivery_carrier_label_dispatch

- change name to batch to match with change of `picking_dispatch` name to
`stock_batch_picking`
- mode description in README.rst
- move files in models and views directories
- change headers to small license header
- use api 8.0
- fix and improve tests

yvaucher committed:
Update translations

yvaucher committed:
Update translations

yvaucher committed:
Cannot update options due to limitation of api 8.0

yvaucher committed:
Cannot update options due to limitation of api 8.0

mdietrichc2c committed:
Use IDs instead of recordset for write

mdietrichc2c committed:
Use IDs instead of recordset for write

Commit:
rename field tracking_id to package_id

the field has been renamed in the base module

Commit:
rename field tracking_id to package_id

the field has been renamed in the base module

mdietrichc2c committed:
Remove self.pool + add correct attachment name

mdietrichc2c committed:
Remove self.pool + add correct attachment name

mdietrichc2c committed:
View changed in batch_stock_picking

mdietrichc2c committed:
View changed in batch_stock_picking

Commit:
Fix condition

Commit:
Fix condition

Commit:
Thread label generation to make it faster

Port of https://github.com/OCA/delivery-carrier/pull/27

Commit:
Thread label generation to make it faster

Port of https://github.com/OCA/delivery-carrier/pull/27

mdietrichc2c committed:
Fix onchange for batch picking

mdietrichc2c committed:
Fix onchange for batch picking

mdietrichc2c committed:
Correct error treatment

mdietrichc2c committed:
Correct error treatment

Commit:
Fix empty pdf issue

Probably due to usage of multiple environment and cache. datas can't be
read from shipping.label but it can be read from ir.attachment

Commit:
Fix empty pdf issue

Probably due to usage of multiple environment and cache. datas can't be
read from shipping.label but it can be read from ir.attachment

mdietrichc2c committed:
Add pick/pack info to all exceptions

mdietrichc2c committed:
Add pick/pack info to all exceptions

mdietrichc2c committed:
e.message deprecated from 2.6

mdietrichc2c committed:
e.message deprecated from 2.6

Commit:
Allow to print labels on operations without product

Commit:
Allow to print labels on operations without product

Commit:
Fix concurrent error on batch labels

When there is more than 1 pack in a picking, 2 or more thread worker
might be in concurrence on the same picking, which makes one of the
transaction fails, resulting in the abort of the whole label generation.

The solution implemented is to change the way of passing the tasks to
the thread workers. Instead of passing a label at a time, we pass a
group of label. A group of label actually contains all the labels to
generate for one picking.

Commit:
Fix concurrent error on batch labels

When there is more than 1 pack in a picking, 2 or more thread worker
might be in concurrence on the same picking, which makes one of the
transaction fails, resulting in the abort of the whole label generation.

The solution implemented is to change the way of passing the tasks to
the thread workers. Instead of passing a label at a time, we pass a
group of label. A group of label actually contains all the labels to
generate for one picking.

Commit:
Do not regenerate existing labels unless forced + pylint fix

Commit:
Do not regenerate existing labels unless forced + pylint fix

TDu committed:
[12.0][MIG] delivery_carrier_label_batch

TDu committed:
[12.0][MIG] delivery_carrier_label_batch

Commit:
[9.0][ADD] support of zpl labels

Commit:
[9.0][ADD] support of zpl labels

sebalix committed:
[IMP] delivery_carrier_label_batch: add support for ZPL labels

sebalix committed:
[IMP] delivery_carrier_label_batch: add support for ZPL labels

tschanzt committed:
fixes for cahnges in migration of base_deliver_carrier_label (#1)

tschanzt committed:
fixes for cahnges in migration of base_deliver_carrier_label (#1)

Commit:
Fix tests

Commit:
Fix tests

Commit:
add missing _description

Commit:
add missing _description

grindtildeath committed:
[IMP] delivery_carrier_label_batch: ZPL2 single images

grindtildeath committed:
[IMP] delivery_carrier_label_batch: ZPL2 single images

grindtildeath committed:
Apply suggestions from code review

Co-Authored-By: Alexandre Fayolle <alexandre.fayolle@camptocamp.com>

grindtildeath committed:
Apply suggestions from code review

Co-Authored-By: Alexandre Fayolle <alexandre.fayolle@camptocamp.com>

grindtildeath committed:
Fixes after reviews

grindtildeath committed:
Fixes after reviews

vrenaville committed:
[IMP][ZPL] option to not merge Zpl file (printer failed to print to big files)

vrenaville committed:
[IMP][ZPL] option to not merge Zpl file (printer failed to print to big files)

vrenaville committed:
[IMP] label name from batch tracking zpl

vrenaville committed:
[IMP] label name from batch tracking zpl

yvaucher committed:
Add a check that all picking have packs before generating them.

yvaucher committed:
Add a check that all picking have packs before generating them.

yvaucher committed:
label batch: Enforce use of packages

The support of picking without packages has been dropped following the RFC #192

Since ffaab97 commit `base_delivery_carrier_label` force the usage of
packages.

yvaucher committed:
label batch: Enforce use of packages

The support of picking without packages has been dropped following the RFC #192

Since ffaab97 commit `base_delivery_carrier_label` force the usage of
packages.

yvaucher committed:
Fix rst format, not an inline emphasis

yvaucher committed:
Fix rst format, not an inline emphasis

oca-travis committed:
[UPD] Update delivery_carrier_label_batch.pot

oca-travis committed:
[UPD] Update delivery_carrier_label_batch.pot

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[ADD] icon.png

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

yvaucher committed:
Remove broken icon

yvaucher committed:
Remove broken icon

OCA-git-bot committed:
delivery_carrier_label_batch 12.0.1.0.1

OCA-git-bot committed:
delivery_carrier_label_batch 12.0.1.0.1

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_carrier_label_batch/

oca-transbot committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-12.0/delivery-carrier-12.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-12-0/delivery-carrier-12-0-delivery_carrier_label_batch/

kv1612 committed:
[IMP] delivery_carrier_label_batch: black, isort, prettier

kv1612 committed:
[IMP] delivery_carrier_label_batch: black, isort, prettier

kv1612 committed:
[MIG] delivery_carrier_label_batch: Migration to 14.0

kv1612 committed:
[MIG] delivery_carrier_label_batch: Migration to 14.0

oca-travis committed:
[UPD] Update delivery_carrier_label_batch.pot

oca-travis committed:
[UPD] Update delivery_carrier_label_batch.pot

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

OCA-git-bot committed:
[IMP] update dotfiles [ci skip]

grindtildeath committed:
delivery_carrier_label_batch: Allow testing labels generation

The threading option doesn't work in test mode but was not tested
because labels were already generated in the test.
Using a condition to use different threads only when test mode
is not activated allows to test label generation.

grindtildeath committed:
delivery_carrier_label_batch: Allow testing labels generation

The threading option doesn't work in test mode but was not tested
because labels were already generated in the test.
Using a condition to use different threads only when test mode
is not activated allows to test label generation.

grindtildeath committed:
Apply suggestions from code review

Co-authored-by: Hai Lang <hailn@trobz.com>

grindtildeath committed:
Apply suggestions from code review

Co-authored-by: Hai Lang <hailn@trobz.com>

sebalix committed:
delivery_carrier_label_batch: empty cache once threads finished

Commit:
delivery_carrier_label_batch: empty cache once threads finished

Commit:
[IMP] delivery_carrier_label_batch: tracking reference purge

Sometimes, when regenerating labels from third party API, we could not get the
new tracking references. And then as we do not purge tracking references
correctly, previous ones are kept and lead to incoherences (batch picking
tracking references emptied but labels still existing). For this we need to
manually purge references when asking for full regeneration.

Commit:
[IMP] delivery_carrier_label_batch: tracking reference purge

Sometimes, when regenerating labels from third party API, we could not get the
new tracking references. And then as we do not purge tracking references
correctly, previous ones are kept and lead to incoherences (batch picking
tracking references emptied but labels still existing). For this we need to
manually purge references when asking for full regeneration.

Commit:
FIX delivery_carrier_label_batch invalid field

Commit:
FIX delivery_carrier_label_batch invalid field

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.0.1

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.0.1

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.0.2

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.0.2

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.1.0

OCA-git-bot committed:
delivery_carrier_label_batch 14.0.1.1.0

OCA-git-bot committed:
[UPD] README.rst

OCA-git-bot committed:
[UPD] README.rst

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/

weblate committed:
Update translation files

Updated by "Update PO files to match POT (msgmerge)" hook in Weblate.

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/

pedrobaeza committed:
[FIX] *: Linter fixes

pedrobaeza committed:
[FIX] *: Linter fixes

OCA-git-bot committed:
[BOT] post-merge updates

OCA-git-bot committed:
[BOT] post-merge updates

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Added translation using Weblate (Spanish)

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 2.8% (1 of 35 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 2.8% (1 of 35 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (35 of 35 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/es/

Ivorra78 committed:
Translated using Weblate (Spanish)

Currently translated at 100.0% (35 of 35 strings)

Translation: delivery-carrier-14.0/delivery-carrier-14.0-delivery_carrier_label_batch
Translate-URL: https://translation.odoo-community.org/projects/delivery-carrier-14-0/delivery-carrier-14-0-delivery_carrier_label_batch/es/

Commit:
delivery_carrier_label_batch: fix concurrent update

Regenerating the carrier labels would crash with a Concurent Update
error.

We use a new cursor for tracking reference purge and avoid concurrent update
when the labels regeneration also updates that field in a separate
cursor afterwards: this ensures that the purge is committed before the
transactions created by the threads also update and commit their own
changes.

Commit:
delivery_carrier_label_batch: fix concurrent update

Regenerating the carrier labels would crash with a Concurent Update
error.

We use a new cursor for tracking reference purge and avoid concurrent update
when the labels regeneration also updates that field in a separate
cursor afterwards: this ensures that the purge is committed before the
transactions created by the threads also update and commit their own
changes.

mymage committed:
Added translation using Weblate (Italian)

mymage committed:
Added translation using Weblate (Italian)

yankinmax committed:
[IMP] delivery_carrier_label_batch: pre-commit auto fixes

yankinmax committed:
[IMP] delivery_carrier_label_batch: pre-commit auto fixes

yankinmax committed:
[MIG] delivery_carrier_label_batch: Migration to 18.0

yankinmax committed:
[MIG] delivery_carrier_label_batch: Migration to 18.0

yankinmax commented:
The tests are failing because of the issue I've mentioned in this PR (`delivery_postlogistics`)
- [ ] https://github.com/OCA/delivery-carrier/pull/974

imlopes commented on delivery_carrier_label_batch/models/stock_batch_picking.py:
```diff
@@ -0,0 +1,129 @@
+
+    carrier_id = fields.Many2one(
+        comodel_name="delivery.carrier",
+        string="Carrier",
```
Redundant `string`

yankinmax replied:
why redundant? as I know many2x fields aren't handled automatically. Am I wrong?

vvrossem replied:
Odoo defaults to "Carrier" by formatting the technical name (`carrier_id`): https://github.com/odoo/odoo/blob/18.0/odoo/fields.py#L503-L508

imlopes commented on delivery_carrier_label_batch/models/stock_batch_picking.py:
```diff
@@ -0,0 +1,129 @@
+    )
+    option_ids = fields.Many2many(
+        comodel_name="delivery.carrier.option",
+        string="Options",
```
Idem above

rousseldenis commented:
/ocabot migration delivery_carrier_label_batch

vvrossem commented on delivery_carrier_label_batch/data/ir.config_parameter.xml:
```diff
@@ -1,10 +1,14 @@
 <odoo noupdate="1">
-    <record id="zpl2_assembler_single_images" model="ir.config_parameter">
+    <record
+        forcecreate="True"
```
Why adding `forcecreate`? 
I'm asking because it is unclear to me when to use it

yankinmax replied:
As I understand if the record doesn't exist `forcecreate` ensures it will be created. I think to use the module features we need parameter to be 100% created.

vvrossem commented on delivery_carrier_label_batch/models/stock_batch_picking.py:
```diff
@@ -36,37 +51,29 @@ def _get_options_to_add(self, carrier=None):
                 res["warning"] = {
-                    "title": _("User Error !"),
-                    "message": _(
+                    "title": self.env._("User Error !"),
+                    "message": self.env._(
```
nitpicking: use `raise UserError`

yankinmax committed:
DON'T MERGE test requirements

yankinmax committed:
DON'T MERGE test requirements

Ricardoalso commented on delivery_carrier_label_batch/wizard/generate_labels_view.xml:
```diff
@@ -0,0 +1,46 @@
+        <field name="view_mode">form</field>
+        <field name="view_id" ref="view_delivery_carrier_label_generate" />
+        <field
+            name="groups_id"
+            eval="[Command.set([ref('stock.group_stock_manager')])]"
```
_Any view with groups_id on it now has to move such groups to the elements of the view. But now you can put groups attribute in a view field instead of isolating it on a view without fear of an access error. Check it in https://github.com/odoo/odoo/pull/98551._

yankinmax replied:
It's not a group.
It's `<record id="action_delivery_carrier_label_generate" model="ir.actions.act_window">`
`groups_id` is still relevant for the window actions
I know this core change

Ricardoalso replied:
Indeed, I didn't check the model. 

Sorry for the noise

Ricardoalso commented on delivery_carrier_label_batch/views/stock_batch_picking.xml:
```diff
@@ -0,0 +1,34 @@
+                        <field name="carrier_id" readonly="state == 'done'" />
+                    </group>
+                    <separator string="Options" />
+                    <field name="option_ids_domain" invisible="1" />
```
_Missing fields used in domains and other attributes are added to the view as invisible automatically. If you encounter fields in views with invisible="True", they can probably be removed. Reference: https://github.com/odoo/odoo/pull/137031_

yankinmax replied:
Thanks for suggestion

StephaneMangin commented:
`delivery_postlogistics` is a final module, why do you need to inherit from it?

yankinmax commented:
@StephaneMangin if the field `parcel_tracking = fields.Char()` is defined in postlogistics I need to inherit

yankinmax commented:
**NOTE for reviewers**
Tests are failing and BTW not even run, because of `Deprecation Warnings` coming from `delivery_postlogistics`.
Locally tested and should be ok.

Camille0907 commented:
FTR the force-push made today adds [this commit](https://github.com/OCA/delivery-carrier/pull/984/commits/2cf903e4052f502817c67a5b70c3fa285948ff0e) to fix an assignation to a field that doesn't exist (anymore ?)

StephaneMangin commented:
Hi @yankinmax How are you ? You can pull this dependency to avoid depending from postlogistics https://github.com/OCA/delivery-carrier/pull/983
I'll manage a PR to remove this field from postlogistics. For now it should not be a problem to have this field on two different modules if the signature of the field is the same during the change (should be though, like so `parcel_tracking = fields.Char()`)

yankinmax commented:
> Hi @yankinmax How are you ? You can pull this dependency to avoid depending from postlogistics #983 I'll manage a PR to remove this field from postlogistics. For now it should not be a problem to have this field on two different modules if the signature of the field is the same during the change (should be though, like so `parcel_tracking = fields.Char()`)

Hello @StephaneMangin , I'm fine, thanks, hope you're too :smile:
Do I correctly understand you: you suggest removing `delivery_postlogitics` dependency in favor of `delivery_carrier_package_info`?
It looks logical for me.
I've updated my migration.

StephaneMangin commented:
> > Hi @yankinmax How are you ? You can pull this dependency to avoid depending from postlogistics #983 I'll manage a PR to remove this field from postlogistics. For now it should not be a problem to have this field on two different modules if the signature of the field is the same during the change (should be though, like so `parcel_tracking = fields.Char()`)
> 
> Hello @StephaneMangin , I'm fine, thanks, hope you're too 😄 Do I correctly understand you: you suggest removing `delivery_postlogitics` dependency in favor of `delivery_carrier_package_info`? It looks logical for me. I've updated my migration.

Yes that's it. Thanks

Camille0907 commented:
> FTR the force-push made today adds [this commit](https://github.com/OCA/delivery-carrier/pull/984/commits/2cf903e4052f502817c67a5b70c3fa285948ff0e) to fix an assignation to a field that doesn't exist (anymore ?)

[EDIT] Re-added [here](https://github.com/OCA/delivery-carrier/pull/984/commits/18449d54a87f0c5c7d1c6460c70469e8ee737565)

Hello, the commit mentioned in my previous message disappeared, probably during a force-push :thinking: 
I'm about to re-add it

Please try to keep it during next force-push :wink:

Camille0907 committed:
[18.0] [FIX] stock_batch_picking.py: remove line updating `self.carrier_code`

    This line has been added by https://github.com/OCA/delivery-carrier/commit/3b0232dae3aa5073d69e18b593675a8bdd557ad9 but it looks like the field carrier_code has never been added to the model (or at least it has never been merged in the OCA repo?), thus we get following error when updating the ``carrier_id``:
    ```
    [...]
      File "/odoo/src/addons/web/models/models.py", line 1005, in onchange
        record._apply_onchange_methods(field_name, result)
      File "/odoo/src/odoo/models.py", line 7364, in _apply_onchange_methods
        res = method(self)
              ^^^^^^^^^^^^
      File "/odoo/external-src/delivery-carrier/delivery_carrier_label_batch/models/stock_batch_picking.py", line 61, in onchange_carrier_id
        self.carrier_code = self.carrier_id.code
        ^^^^^^^^^^^^^^^^^
    AttributeError: 'stock.picking.batch' object has no attribute 'carrier_code'

    The above server error caused the following client error:
    RPC_ERROR: Odoo Server Error
        RPC_ERROR
            at makeErrorFromResponse (http://localhost:8069/web/assets/e5ed061/web.assets_web.min.js:3144:163)
            at XMLHttpRequest.<anonymous> (http://localhost:8069/web/assets/e5ed061/web.assets_web.min.js:3149:13)
    ```

Camille0907 committed:
[18.0] [FIX] stock_batch_picking.py: remove line updating `self.carrier_code`

    This line has been added by https://github.com/OCA/delivery-carrier/commit/3b0232dae3aa5073d69e18b593675a8bdd557ad9 but it looks like the field carrier_code has never been added to the model (or at least it has never been merged in the OCA repo?), thus we get following error when updating the ``carrier_id``:
    ```
    [...]
      File "/odoo/src/addons/web/models/models.py", line 1005, in onchange
        record._apply_onchange_methods(field_name, result)
      File "/odoo/src/odoo/models.py", line 7364, in _apply_onchange_methods
        res = method(self)
              ^^^^^^^^^^^^
      File "/odoo/external-src/delivery-carrier/delivery_carrier_label_batch/models/stock_batch_picking.py", line 61, in onchange_carrier_id
        self.carrier_code = self.carrier_id.code
        ^^^^^^^^^^^^^^^^^
    AttributeError: 'stock.picking.batch' object has no attribute 'carrier_code'

    The above server error caused the following client error:
    RPC_ERROR: Odoo Server Error
        RPC_ERROR
            at makeErrorFromResponse (http://localhost:8069/web/assets/e5ed061/web.assets_web.min.js:3144:163)
            at XMLHttpRequest.<anonymous> (http://localhost:8069/web/assets/e5ed061/web.assets_web.min.js:3149:13)
    ```

yankinmax commented:
Sorry about it. It’s definitely accidentally. I’ll be more attentive. Sorry
again
пт, 13 черв. 2025 р. о 17:43 Camille Morand ***@***.***> пише:

> *Camille0907* left a comment (OCA/delivery-carrier#984)
> <https://github.com/OCA/delivery-carrier/pull/984#issuecomment-2970625576>
>
> FTR the force-push made today adds this commit
> <https://github.com/OCA/delivery-carrier/pull/984/commits/2cf903e4052f502817c67a5b70c3fa285948ff0e>
> to fix an assignation to a field that doesn't exist (anymore ?)
>
> Hello, the commit mentioned in my previous message disappeared, probably
> during a force-push 🤔
> I'm about to re-add it
>
> Please try to keep it during next force-push 😉
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/OCA/delivery-carrier/pull/984#issuecomment-2970625576>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AG6ZRRXGU2EG4ECZDPP5QXT3DLPQJAVCNFSM6AAAAABXEHNPBCVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDSNZQGYZDKNJXGY>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

Camille0907 committed:
[18.0] [FIX] delivery_carrier_label_batch: replace ``UserError.name`` by ``UserError.args[0]``

In the label generation, when getting an error, we were using its ``name``, however, this is deprecated since a while and as been removed in https://github.com/odoo/odoo/commit/d200dcfb2c912353d404312b9d9484846868cf96.
We should use ``args[0]`` instead.

Camille0907 committed:
[18.0] [FIX] delivery_carrier_label_batch: replace ``UserError.name`` by ``UserError.args[0]``

In the label generation, when getting an error, we were using its ``name``, however, this is deprecated since a while and as been removed in https://github.com/odoo/odoo/commit/d200dcfb2c912353d404312b9d9484846868cf96.
We should use ``args[0]`` instead.

Camille0907 commented:
Hello, my new force-push is to add another commit: https://github.com/OCA/delivery-carrier/pull/984/commits/f0f852e61b836b316247f13dd96fa6175fe08929

It fixes a traceback appearing when we manage errors happening during label generation 

We might want to squash it with the migration commit ?

Camille0907 committed:
[18.0] [FIX] delivery_carrier_label_batch: make ``_get_packs`` method return ``stock.move.line`` records instead of an Iterator

All calls to this method expect ``stock.move.line`` records.

Camille0907 committed:
[18.0] [FIX] delivery_carrier_label_batch: make ``_get_packs`` method return ``stock.move.line`` records instead of an Iterator

All calls to this method expect ``stock.move.line`` records.

gurneyalex committed:
[IMP] delivery_carrier_batch: limit number of threads

Don't spawn more threads than needed: if there are fewer groups to
process than the configured max number of threads, only start the
specified number of threads.

gurneyalex committed:
[IMP] delivery_carrier_batch: limit number of threads

Don't spawn more threads than needed: if there are fewer groups to
process than the configured max number of threads, only start the
specified number of threads.

gurneyalex commented:
@yankinmax @Camille0907 I just pushed 2 new commits on this PR

yankinmax commented:
> @yankinmax @Camille0907 I just pushed 2 new commits on this PR

Thanks @gurneyalex

yankinmax commented on delivery_carrier_label_batch/wizard/generate_labels.py:
```diff
@@ -0,0 +1,342 @@
+            )
+            raise exceptions.UserError(msg)
+
+    def action_generate_labels(self):
```
For some reason, the labels generated aren't ordered as expected.
Does anyone using this module have the same issue?

grindtildeath commented:
@yankinmax @wouitmil 
Can one of you guys please clean up that PR and make github actions happy? :pray:

cyrilmanuel commented on test-requirements.txt:
```diff
@@ -1,2 +1,4 @@
 odoo-test-helper
 vcrpy
+odoo-addon-delivery_carrier_package_info @ git+https://github.com/OCA/delivery-carrier.git@refs/pull/983/head#subdirectory=delivery_carrier_package_info
+odoo-addon-stock_picking_batch_extended @ git+https://github.com/OCA/stock-logistics-workflow.git@refs/pull/1855/head#subdirectory=stock_picking_batch_extended
```
pr closed and superssed on https://github.com/OCA/stock-logistics-workflow/pull/2120

cyrilmanuel commented on test-requirements.txt:
```diff
@@ -1,2 +1,4 @@
 odoo-test-helper
 vcrpy
+odoo-addon-delivery_carrier_package_info @ git+https://github.com/OCA/delivery-carrier.git@refs/pull/983/head#subdirectory=delivery_carrier_package_info
```
https://github.com/OCA/delivery-carrier/pull/983 PR closed need to be superseed

yankinmax commented:
hello @grindtildeath @cyrilmanuel 
I'll take some time to update this PR soon. Thanks

flotho commented:
Hi @yankinmax , any chance to rebase ?
Regards

yankinmax commented:
> Hi @yankinmax , any chance to rebase ? Regards

Rebased
